UK_Power_Generation_Units_BMRS/Python/Code/ukpowersupply/Script-BMRS-API-Marginal-Seller/BMRS-API-JA-MarginalSeller.py

```diff
@@ -0,0 +1,210 @@
+#KEY: iwx6raw9m7nqq0f
+
+#TEST from page 193: https://www.elexon.co.uk/documents/training-guidance/bsc-guidance-notes/bmrs-api-and-data-push-user-guide-2/
+
+###Libraries###
+import httplib2
+from pprint import pformat
+
+import re
+import pandas as pd
+import sys
+from datetime import datetime, timedelta
+
+
+####Functions###
+def str0_2(value):
+    #Converts the day or month int to a string of length 2. Thus, 12 -> "12", and 1 -> "01", the leading 0 is important.
+    #This function thus performs a similar role as str(), but also can add the 0 at the start, and is applied to length 2 instances.
+    #Must use for day or month, but use for period is optional, as length 1 and 2 is accepted by the API format the period, but not the month or day. 
+    value = str(value)
+    if len(value) == 1:
+        value = "0"+value
+    return value
+
+
+def next_day(Year, Month, Day):
+    #Convert to timestamp.
+    day = datetime(int(Year), int(Month), int(Day), 0, 0, 0, 0)
+    
+    #Get a week ago (returns str for year, month, day: 
+    nextDay = day + timedelta(days=1)
+    print(nextDay)
+    return str(nextDay.year), str0_2(nextDay.month), str0_2(nextDay.day)
+
+
+def post_elexon(url):
+    http_obj = httplib2.Http()
+    resp, content = http_obj.request(
+        uri=url,
+        method='GET',
+        headers={'Content-Type': 'application/xml; charset=UTF-8'},
+    )
+    #print('===Response===')
+    #print(pformat(resp))
+    
+    #print('===Content===')
+    #print(pformat(content))
+    #print(type(pformat(content)))
+
+    #print('===Finished===')
+    
+    return pformat(content)
+
+
+def run_price_query(YYYY, MM, DD, PP):
+    #Query All
+    makeurl = 'https://api.bmreports.com/BMRS/B1770/v1?APIKey=iwx6raw9m7nqq0f&SettlementDate='+YYYY+'-'+MM+'-'+DD+'&Period='+PP+'&ServiceType=xml'
```

Comment:
Please remove the API key value from lines like this (there are a few in this file). Don't remove the lines as a whole.